swap.go

```diff
@@ -39,18 +42,25 @@ func newSwapKit(hash lntypes.Hash, swapType swap.Type, cfg *swapConfig,
 	}
 
 	return &swapKit{
-		swapConfig: *cfg,
-		hash:       hash,
-		log:        log,
-		state:      loopdb.StateInitiated,
-		contract:   contract,
-		swapType:   swapType,
+		swapConfig:      *cfg,
+		hash:            hash,
+		log:             log,
+		state:           loopdb.StateInitiated,
+		contract:        contract,
+		swapType:        swapType,
+		protocolVersion: protocolVersion,
 	}
 }
 
 // getHtlc composes and returns the on-chain swap script.
 func (s *swapKit) getHtlc(outputType swap.HtlcOutputType) (*swap.Htlc, error) {
+	scriptVersion := swap.HtlcV1
+	if s.protocolVersion >= loopdb.ProtocolVersionHtlcV2 {
```

Comment:
Won't this set HtlcV2 for s.protocolVersion = `ProtocolVersionUnrecorded`? Which would mean any swap that is resumed after upgrade would have the wrong htlc type.

Reply:
Good catch! This was left when changed to db protocol version from proto protocol version.